pkg/arvo/app/permission-group-hook.hoon

```diff
@@ -0,0 +1,179 @@
+/-  *group-store, *permission-store, *permission-group-hook
+|%
++$  move  [bone card]
+::
++$  card
+  $%  [%diff [%group-update group-update]]
+      [%quit ~]
+      [%poke wire dock poke]
+      [%pull wire dock ~]
+      [%peer wire dock path]
+  ==
+::
++$  state
+  $%  [%0 state-zero]
+  ==
+::
++$  group-path  path
+::
++$  permission-path  path
+::
++$  state-zero
+  $:  relation=(map group-path (set permission-path))
+  ==
+::
++$  poke
+  $%  [%permission-action permission-action]
+  ==
+::
+--
+::
+|_  [bol=bowl:gall state]
+::
+++  this  .
+::
+++  prep
+  |=  old=(unit state)
+  ^-  (quip move _this)
+  ?~  old
+    [~ this]
+  [~ this(+<+ u.old)]
+::
+++  poke-permission-group-hook-action
+  |=  act=permission-group-hook-action
+  ^-  (quip move _this)
+  ?.  =(src.bol our.bol)
+    [~ this]
+  ?-  -.act
+      %associate
+    (handle-associate group.act permissions.act)
+  ::
+      %dissociate
+    (handle-dissociate group.act permissions.act)
+  ::
+  ==
+::
+++  handle-associate
+  |=  [group=path permissions=(set path)]
+  ^-  (quip move _this)
+  ::  TODO: remove the requirement for permission paths to exist
+  =/  perm-list  ~(tap in permissions)
+  =/  perm-length  (lent perm-list)
+  =/  checked-paths
+    %+  skim  perm-list
+    |=  pax=path
+    ^-  ?
+    ?~  (permission-scry pax)
+      %.n
+    %.y
+  ?.  =((lent checked-paths) perm-length)
+    [~ this]
+  =/  perms  (~(get by relation) group)
```

Comment:
This is also a jug operation, not sure whether it's clearer:

```
  =/  perms  (~(get ju relation) group)
  ::  if relation does not exist, create it and subscribe.
  :-  ?^  perms
        ~
      =/  group-path  [%group group]
      [ost.bol %peer group-path [our.bol %group-store] group-path]~
  =.  perms  (~(uni in perms) permissions)
  this(relation (~(put by relation) group perms))
```

Reply:
I need to refactor this function anyway. I have a strong aesthetic distaste for jugs, but I agree the logic of this function could be simplified.

Reply:
I don't feel strongly about jugs

Reply:
Finished the refactor. Feels nicer without need for this.